### PR TITLE
feat(stepfunctions-tasks): `algorithmName` validation for `SageMakerCreateTrainingJob`

### PR DESCRIPTION
**Pull request by tmyoda**
Referencing PR #26675, I have added validation for the `algorithmName` parameter in `SageMakerCreateTrainingJob`.
However, it was suggested that changes for validation should be separated.  So, I have created this PR.

Docs for `algorithmName`:
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_AlgorithmSpecification.html#API_AlgorithmSpecification_Contents

Exemption Request:
This change does not alter the behavior.
I believe the unit test `create-training-job.test.ts` that I have added is sufficient to test this change.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by tmyoda** [comment]
Exemption Request:
This change does not alter the behavior.
I believe the unit test `create-training-job.test.ts` that I have added is sufficient to test this change.

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts [hunk 1]
```diff
@@ -324,6 +332,21 @@ export class SageMakerCreateTrainingJob extends sfn.TaskStateBase implements iam
       : {};
   }
 
+  private validateAlgorithmName(algorithmName?: string): void {
+    if (algorithmName === undefined) {
+      return;
+    }
```
Comment (rix0rrr) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts:338]: This also needs check for `Token.isUnresolved()`.

**Review by rix0rrr — CHANGES_REQUESTED** [review]
Almost there!

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts (hunk 1 above)
Comment (tmyoda) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts]: I added check for `Token.isUnresolved()`

**Review by rix0rrr — DISMISSED** [review]

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**